app/controllers/api/v2/config_templates_controller.rb

```diff
@@ -79,9 +79,10 @@ def destroy
         process_response @config_template.destroy
       end
 
-      api :GET, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
+      api :POST, "/config_templates/build_pxe_default", N_("Update the default PXE menu on all configured TFTP servers")
 
       def build_pxe_default
+        ::ActiveSupport::Deprecation.warn('GET method for build pxe default is deprecated. Please use POST instead.') if request.method == "GET"
```

Comment:
I think it's ok to document only the valid option and hide the deprecated one. Existing scripts will keep working and new users will use POST after reading the docs.